rasa/core/training/interactive.py

```diff
@@ -584,14 +610,37 @@ def _slot_history(tracker_dump: Dict[Text, Any]) -> List[Text]:
         for a in predictions
     ]
 
+    tracker = await retrieve_tracker(endpoint, sender_id)
+    evts = tracker.get("events", [])
+
+    session_actions_all = [a["name"] for a in _collect_actions(evts)]
+    session_actions_unique = list(set(session_actions_all))
+    old_actions = [action["value"] for action in choices]
+    new_actions = [
+        {"name": action, "value": OTHER_ACTION + action}
+        for action in session_actions_unique
+        if action not in old_actions
+    ]
     choices = [{"name": "<create new action>", "value": OTHER_ACTION}] + choices
     question = questionary.select("What is the next action of the bot?", choices)
 
     action_name = await _ask_questions(question, sender_id, endpoint)
-    is_new_action = action_name == OTHER_ACTION
+    is_new_action = action_name == NEW_ACTION
 
     if is_new_action:
+        # create new action
         action_name = await _request_free_text_action(sender_id, endpoint)
+        if "utter_" in action_name:
+            utter_message = await _request_free_text_utterance(
+                sender_id, endpoint, action_name
+            )
+            NEW_TEMPLATES[action_name] = {utter_message: utter_message}
```

Comment:
ok i'm still not sure why we have an `utter_message: utter_message` here. It should surely be something like `{"text": utter_message}`?

Reply:
https://github.com/RasaHQ/rasa_nlu/pull/3206#discussion_r274140410 
The key and not the value is actually picked up for writing the utter template to the domain file, this is why I put the message "template from IL" before. This is done to have the same dictionary format for all templates when reading them. If the message were to include buttons they'd go in as the value
```
utter_ask_color:
- text: "what color would you like?"
  buttons:
  - title: "blue"
    payload: '/choose{"color": "blue"}'
  - title: "red"
    payload: '/choose{"color": "red"}'
```
while the text is always the key.

Reply:
ahhh, ok let's maybe just make it `""` then